main/windows/index.ts

```diff
@@ -554,35 +554,59 @@ const init = () => {
 
   store.observer(() => broadcast('permissions', JSON.stringify(store('permissions'))))
   store.observer(() => {
-    const summonShortcut = store('main.shortcuts.summon')
-    const accelerator = getAcceleratorFromShortcut(summonShortcut)
-    try {
-      globalShortcut.unregister(accelerator)
-      if (summonShortcutAccelerator) {
-        globalShortcut.unregister(summonShortcutAccelerator)
+    const keyboardLayout = store('keyboardLayout')
+    let summonShortcut: Shortcut = store('main.shortcuts.summon')
+    const summonHandler = (accelerator: string) => {
+      app.toggle()
+      if (store('windows.onboard.showing')) {
+        send('onboard', 'main:flex', 'shortcutActivated')
       }
-      if (summonShortcut.enabled && !summonShortcut.configuring) {
-        globalShortcut.register(accelerator, () => {
-          app.toggle()
-          if (store('windows.onboard.showing')) {
-            send('onboard', 'main:flex', 'shortcutActivated')
-          }
+      if (tray?.isReady()) {
+        systemTray.setContextMenu(tray.isVisible() ? 'hide' : 'show', {
+          displaySummonShortcut: summonShortcut.enabled,
+          accelerator
         })
-        summonShortcutAccelerator = accelerator
       }
-    } catch (e) {
-      const summonShortcutStr = [...summonShortcut.modifierKeys, summonShortcut.shortcutKey].join('+')
-      log.error(
-        new Error(`Could not set accelerator "${accelerator}" for summon shortcut: ${summonShortcutStr}`)
-      )
     }
 
-    if (tray?.isReady()) {
-      systemTray.setContextMenu(tray.isVisible() ? 'hide' : 'show', {
-        displaySummonShortcut: summonShortcut.enabled,
-        accelerator
-      })
+    // Windows & Linux Non-US key layout AltGr / Right Alt fix
+    const createSummonShortcutNonUS = () => {
+      // remove AltGr and Alt from modifiers
+      const modifierKeys = summonShortcut.modifierKeys.filter((modifier) => !modifier.startsWith('Alt'))
+
+      // return new modifiers depending on OS + rest of shortcut - so that AltGr / Right Alt triggers summon in the same way as Left Alt
+      return {
```

Comment:
is it possible to extract some of the logic from here down out and write some unit tests to demonstrate what happens in different scenarios? it's a bit hard to follow what changes we're actually making to the shortcuts at fist glance here

Reply:
Yeah, I think this would improve with another level of extraction - such would be necessary in the case of us using more than one keyboard shortcut, I just took the path of YAGNI here given that we are only using one shortcut atm.